code/game/objects/items/weapons/soap.dm

```diff
@@ -12,6 +12,10 @@
 	throw_range = 20
 	discrete = 1
 	var/cleanspeed = 50 //slower than mop
+	var/times_eaten = 0 //How many times a Drask has chewed on this bar of soap
+	var/max_bites = 30 //The maximum amount of bites before the soap is depleted
+	var/overeat_counter = 0 //How many times this bar of soap has been RECENTLY chewed on.
+	var/overeat_limit = 7 //If you've eaten soap this many times in the last 20 seconds, you get a tummyache
```

Comment:
I think `overeat_limit` might be better as a define, I think `max_bites` is better as a non-define though so that you can adminbus for infinite soap